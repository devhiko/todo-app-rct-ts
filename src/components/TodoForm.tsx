import '../styles/TodoForm.css'
import { useState } from 'react'
import { v1 } from 'uuid'

const TodoForm = ({ setTodos }) => {
  const [todoItem, setTodoItem] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    const newTodo = todoItem;
    setTodos((prevTodos) => [...prevTodos, { name: newTodo, id: v1() }]);
    setTodoItem('');
  }
  return (
    <form
      className="todo-form"
      onSubmit={handleSubmit}
    >
      <input
        id='td'
        type="text"
        required
        value={todoItem}
        onChange={(e) => setTodoItem(e.target.value)}
        placeholder="What should be done" />
      <input
        type="submit"
        value="Add Todo" />
    </form>
  );
}

export default TodoForm;