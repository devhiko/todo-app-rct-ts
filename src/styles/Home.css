.Home {
  padding: 10px;
}

.Home ul {
  margin: 0;
  padding: 0;
}

.Home li {
  margin: 10px 5px;
  padding: 10px;
  background: wheat;
}

.Home li:hover {
  text-decoration: line-through;
}