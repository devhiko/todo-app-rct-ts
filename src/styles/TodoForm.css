.todo-form {
  padding: 10px;
}

.todo-form input {
  background: wheat;
  font-size: 16px;
  margin: 5px 10px;
  padding: 10px;
  border: none;
}

.todo-form input[type='submit'] {
  background: coral;
}

.todo-form input[type='submit']:hover {
  background: darkgoldenrod;
}