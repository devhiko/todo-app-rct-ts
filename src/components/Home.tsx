import '../styles/Home.css'
import { useState } from 'react';
import TodoForm from './TodoForm';
import TodoList from './TodoList';

export interface Todo {
  name: string
  id: string
}

const Home = () => {

  const [todos, setTodos] = useState<Todo[]>([]);

  return (
    <div className="Home">
      {todos.length ?
        <TodoList todos={todos} setTodos={setTodos} />
        :
        <p style={{ color: 'wheat' }}>No Todos, have fun !</p>}
      <TodoForm setTodos={setTodos} />
    </div>
  );
}

export default Home;