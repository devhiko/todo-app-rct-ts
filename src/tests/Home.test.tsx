import { render, screen } from "@testing-library/react";
import Home from "../components/Home";
import TodoForm from "../components/TodoForm";

describe("Home test", () => {
  it("should render Home", () => {
    render(<Home />);
    screen.debug();
  });

  it("should have home classname", () => {
    render(<Home />);
    const home = screen.getByTitle("home");
    expect(home).toHaveClass("Home");
  });

  it("should render todo form everytime", () => {
    const mocksetTodos = jest.fn();
    render(<TodoForm setTodos={mocksetTodos} />);
    const form = screen.getByTitle(/form/i);
    expect(form).toBeInTheDocument();
    const input = screen.getByPlaceholderText(/what should/i);
    expect(input).toBeInTheDocument();
    const btn = screen.getByDisplayValue(/add todo/i);
    expect(btn).toBeInTheDocument();
  });

  it("should render no todos text when no todos", () => {
    render(<Home />);
    const todotext = screen.getByText(/no todos/i);
    expect(todotext).toBeInTheDocument();
  });
});
