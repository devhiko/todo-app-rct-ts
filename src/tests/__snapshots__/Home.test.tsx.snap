// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test should have home classname 1`] = `
<div
  class="Home"
  title="home"
>
  <form
    class="todo-form"
    title="form"
  >
    <input
      id="td"
      placeholder="What should be done"
      required=""
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Add Todo"
    />
  </form>
  <p
    style="color: wheat;"
  >
    No Todos, have fun !
  </p>
</div>
`;
